lemmy_api/src/report.rs

```diff
@@ -0,0 +1,394 @@
+use actix_web::web::Data;
+
+use lemmy_db::{
+  comment_report::*,
+  comment_view::*,
+  community_view::*,
+  post_report::*,
+  post_view::*,
+  Reportable,
+  user_view::UserView,
+};
+use lemmy_structs::{blocking, report::*};
+use lemmy_utils::{APIError, ConnectionId, LemmyError};
+use lemmy_websocket::LemmyContext;
+
+use crate::{check_community_ban, get_user_from_jwt, Perform};
+
+const MAX_REPORT_LEN: usize = 1000;
+
+#[async_trait::async_trait(?Send)]
+impl Perform for CreateCommentReport {
+  type Response = CommentReportResponse;
+
+  async fn perform(
+    &self,
+    context: &Data<LemmyContext>,
+    _websocket_id: Option<ConnectionId>,
+  ) -> Result<CommentReportResponse, LemmyError> {
+    let data: &CreateCommentReport = &self;
+    let user = get_user_from_jwt(&data.auth, context.pool()).await?;
+    
+    // Check size of report and check for whitespace
+    let reason: Option<String> = match data.reason.clone() {
+      Some(s) if s.trim().is_empty() => None,
+      Some(s) if s.len() > MAX_REPORT_LEN => {
+        return Err(APIError::err("report_too_long").into());
+      }
+      Some(s) => Some(s),
+      None => None,
+    };
```

Comment:
So you allow reports with an empty reason? I'm not sure if that makes sense.

Reply:
i don't like that either, i will change it